src/components/SRNEntry.vue

```diff
@@ -164,64 +179,87 @@ export default {
         }
-        //only if the SRN is valid or if the user is entering a SRN for the second time,the loop is entered
-        if (this.isUserValid || this.validateCount > 1) {
-          //setting the flag of the SRN
+        if (this.isCurrentUserValid || this.validateCount > 1) {
           this.setValidFlag();
           this.addNewEmptyField();
-          //resetting flags to default for processing next SRN
           this.resetValidFlag;
           this.validateCount = 0;
         }
       }
     },
+    /* Before an element is removed from the userIDList array, the login message and input message are reset to default values. 
+       removeInputField() is called to remove the element. 
+       In most cases, this would be enough. But there is one edge case:
+      Let's say the user typed in ID 'A' and decides to add another entry. This means 'A' is authenticated and it is valid/invalid. 
+      The user types in the ID 'B' as the next entry and it is invalid. 
+      The user changed their mind and decided to delete this entry ('B'). At this point, the variables will be in the following state:
+        validateCount = 1
+        isCurrentUserValid = false
+      The state of the variables contradict the fact that 'A' is already authenticated. This means any of the two cases:
+      - A is valid, so validateCount = 0 and isCurrentUserValid = true
+      - A is invalid, so validateCount = 2 and isCurrentUserValid = false
+      So when the user clicks submit, 'A' is authenticated again, which is a behaviour we do not want. So to bypass this, validateCount is set to 2. 
+      (To better understand how this will help, understanding processForm() will help.) 
+      isCurrentUserValid flag is ignored because the value is already set in each user's valid flag. 
+      @params {Number} - index - index of input field to be removed */
     removeField(index) {
-      //resetting all messages to default before deleting the input field
       this.resetInvalidInputMessage;
       this.resetInvalidLoginMessage;
-      //edge case: user enters an invalid SRN. Error message is displayed. The user is given another chance. User adds another input field but decided to remove it.
-      // At this point, the variables are reset. So the previously entered SRN is checked again, which should not happen. so setting this validateCount = 2 will bypass this.
-      //Will not affect any other case. The user can either submit or decide to add another field again.
       this.validateCount = 2;
       this.removeInputField(index);
     },
+    /*  this function is called whenever something is entered in the input box. It checks if the required number of characters are being typed. 
+    Until then, it prompts the user to type the required characters. 
+    If the user types more than the desired number of characters, the input is sliced and the user wont be able to see the extra characters.
+    @params {Event} - event - the event which triggered this function
```

Comment:
```suggestion
    @param {Object} - event - the event which triggered this function
```
It's just `param` and not `params`